File: hackerrank/min_max/min_max.kt
import java.io.*
import java.util.*

fun main(args: Array<String>) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. */
    val input_string = readLine()!!
    // toLong - read question nicely. Related to overflow in the description. If not it won't pass all cases
    var input_array_sum = input_string.split(" ").map { it.toLong()}.sum()
    var sums = input_string.split(" ").map { input_array_sum - it.toLong() }
    val print_string = String.format("%s %s", sums.min(), sums.max())
    println(print_string)
}

File: hackerrank/min_max/min_max.exs
defmodule Solution do
#Enter your code here. Read input from STDIN. Print output to STDOUT
    input_number = IO.gets ""
    sum = input_number |> String.split 
          |> Enum.map(fn(x) -> String.to_integer(x) end)
          |> Enum.sum
    min_max = input_number |> String.split 
              |> Enum.map(fn(x) -> String.to_integer(x) end)
              |> Enum.map(fn(x) -> sum - x end)
    IO.puts "#{min_max |> Enum.min} #{min_max |> Enum.max}"
end
